### DISCUSSION TITLE
Supporting Java LTS version 11

### DISCUSSION THREAD

**Discussion by hmf** [HanSolo/charts#53] [Q&A] [answered]
I am trying to use the library with a project that compiles for JDK 11. This is an LTS version. Compilation succeeds but execution fails  with:  

```
Exception in thread "JavaFX Application Thread" Exception in thread "main" java.lang.UnsupportedClassVersionError: eu/hansolo/fx/charts/Axis has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```

So [I see](https://javaalmanac.io/bytecode/versions/) that the library has been compiled with JDK16. Short of compiling and installing it locally myself, any way we can get support for earlier LTS versions? Maybe the [versioning numbers](https://search.maven.org/artifact/eu.hansolo.fx/charts/16.0.12/jar) can be changed slightly to reflect this?

On a side note I see [here](https://github.com/HanSolo/charts/blob/master/build.gradle#L32) that source compatibility is set to JDK11. And [here](https://github.com/HanSolo/charts/blob/master/gradle.properties) I see `targetCompatibility = 11`. Maybe I am missing something here, but shouldn't that produce class file versions of 55.0?

TIA

**Comment by HanSolo** [comment, marked as answer]
Hi there, just make sure that you use the right version. The 16.0.x versions need at least JDK16 and are more or less preparation for the next LTS which is JDK17. The 11.x versions are for JDK11 and later.

**Reply by hmf** [reply]
@HanSolo Thanks for the quick reply. Ok, I understand now. Master branch sticks to JDK11 and the jdk16  branch uses JDK 16. I will try to use [version 11.7](https://search.maven.org/artifact/eu.hansolo.fx/charts/11.7/jar).
